src/azure-cli-tests/azure/cli/tests/base.py

```diff
@@ -0,0 +1,167 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+import os.path
+import inspect
+import vcr
+import subprocess
+import json
+import shlex
+
+from azure.cli.tests.patches import (patch_load_cached_subscriptions, patch_main_exception_handler,
+                                     patch_retrieve_token_for_user)
+from azure.cli.tests.exceptions import CliExecutionError
+from azure.cli.tests.const import ENV_LIVE_TEST, ENV_SKIP_ASSERT, MOCKED_SUBSCRIPTION_ID
+from azure.cli.tests.recording_processors import (SubscriptionRecordingProcessor,
+                                                  OAuthRequestResponsesFilter,
+                                                  GeneralNameReplacer)
+from azure.cli.tests.utilities import create_random_name
+
+
+class ScenarioTest(unittest.TestCase):
+    def __init__(self, method_name):
+        super(ScenarioTest, self).__init__(method_name)
+        self.name_replacer = GeneralNameReplacer()
+        self.recording_processors = [SubscriptionRecordingProcessor(MOCKED_SUBSCRIPTION_ID),
+                                     OAuthRequestResponsesFilter(),
+                                     self.name_replacer]
+
+        test_file_path = inspect.getfile(self.__class__)
+        recordings_dir = os.path.join(os.path.dirname(test_file_path), 'recordings')
+        live_test = os.environ.get(ENV_LIVE_TEST, None) == 'True'
+
+        self.vcr = vcr.VCR(
+            cassette_library_dir=recordings_dir,
+            before_record_request=self._process_request_recording,
+            before_record_response=self._process_response_recording,
+            decode_compressed_response=True,
+            record_mode='once' if not live_test else 'all',
+            filter_headers=['authorization'],
+        )
+
+        self.recording_file = os.path.join(recordings_dir, '{}.yaml'.format(method_name))
+        if live_test and os.path.exists(self.recording_file):
+            os.remove(self.recording_file)
+
+        self.skip_assert = os.environ.get(ENV_SKIP_ASSERT, None) == 'True'
+        self.in_recording = live_test or not os.path.exists(self.recording_file)
+        self.test_resources_count = 0
+
+    def setUp(self):
```

Comment:
Is the overriding a base implementation?  If not, it should be in Python casing (set_up)